htlcswitch/spider_test.go

```diff
@@ -6,364 +6,216 @@ import (
-				htlcAmt, totalTimelock, hops := generateHops(amount, testStartingHeight,
-						n.secondBobChannelLink)
-
-				firstHop := n.secondBobChannelLink.ShortChanID()
-				_, err := n.makePayment(
-					n.carolServer, n.bobServer, firstHop, hops, amount, htlcAmt,
-					totalTimelock,
-				).Wait(30 * time.Second)
-
-				if err != nil {
-					t.Fatal("carol->bob failed")
-				}
-				fmt.Println("carol->bob successful try ")
-				time.Sleep(1*time.Second)
-			}
-		}()
-
-		amount := lnwire.NewMSatFromSatoshis(4 * btcutil.SatoshiPerBitcoin)
-		htlcAmt, totalTimelock, hops := generateHops(amount, testStartingHeight,
-					n.firstBobChannelLink, n.carolChannelLink)
-
-		// What we expect to happen:
-		// * HTLC add request to be sent to from Alice to Bob.
-		// * Alice<->Bob commitment states to be updated.
-		// * Bob trying to add HTLC add request in Bob<->Carol channel.
-		// * Not enough funds for that, so it gets added to overFlowQueue in
-		// Bob->Carol channel.
-		// * Carol sends Bob money.
-		// * Bob -> Carol channel now has enough money, and the Bob -> Carol
-		// payment should succeed, thereby letting the Alice -> Carol payment to
-		// succeed as well.
-
-		firstHop := n.firstBobChannelLink.ShortChanID()
-		// launch second payment here, which sleeps for a bit, and then pays from
-		// carol to bob.
-		_, err = n.makePayment(
-			n.aliceServer, n.carolServer, firstHop, hops, amount, htlcAmt,
-			totalTimelock,
-		).Wait(80 * time.Second)
-		// modifications on existing test. We do not want it to fail here.
-		if err != nil {
-			fmt.Println(err)
-			t.Fatal("error has been received in first payment, alice->bob")
-		}
-		fmt.Println("alice -> carol successful")
-		// if we reach this point, then all the payments have succeeded.
-		// Sleep to let all intermediate payments Carol -> bob succeed.
-		time.Sleep(5*time.Second)
+	n, cleanUp := StartThreeHopNetwork(5, 3, t)
+	defer cleanUp()
+	defer n.stop()
+
+	// What we expect to happen:
+	// * HTLC add request to be sent to from Alice to Bob.
+	// * Alice<->Bob commitment states to be updated.
+	// * Bob trying to add HTLC add request in Bob<->Carol channel.
+	// * Not enough funds for that, so it gets added to overFlowQueue in
+	// Bob->Carol channel.
+	// * Carol sends Bob money. Still not enough funds to forward Alice's
+	// request.
+	// * Carol keeps sending Bob more money.
+	// * Eventually, Bob -> Carol channel now has enough money, and the Bob -> Carol
+	// payment should succeed, thereby letting the Alice -> Carol payment to
+	// succeed as well.
+	c := make(chan error)
+	go SendMoneyWithDelay(n, 4 * btcutil.SatoshiPerBitcoin, n.aliceServer, n.carolServer, 0, c, n.firstBobChannelLink, n.carolChannelLink)
+	// carol -> bob after a 2 seconds delay
+	for i := 0; i < 4; i++ {
+		go SendMoneyWithDelay(n, 0.3 * btcutil.SatoshiPerBitcoin, n.carolServer, n.bobServer, int64(i+1), make(chan error), n.secondBobChannelLink)
```

Comment:
Nit: Should the comment reflect that these happen at intervals of 1 second starting from 1 second after the initial payment?